src/pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -463,40 +513,48 @@ def _stop_progress(self) -> None:
             # # signals for progress to be re-initialized for next stages
             self._progress_stopped = True
 
-    def _reset_progress_bar_ids(self):
-        self.main_progress_bar_id: Optional[int] = None
-        self.val_progress_bar_id: Optional[int] = None
-        self.test_progress_bar_id: Optional[int] = None
-        self.predict_progress_bar_id: Optional[int] = None
+    def _reset_progress_bar_ids(self) -> None:
+        self.main_progress_bar_id = None
+        self.val_progress_bar_id = None
+        self.test_progress_bar_id = None
+        self.predict_progress_bar_id = None
```

Comment:
Is it correct that we are removing typing in here?

Reply:
`mypy` always has the issue of not detecting the types of attributed introduced in a class method. I tried to resolve this by introduce these attributes (and their typings) in the `__init__` method instead.

Reply:
Oh! I thought this is in `__init__`! Absolutely fine then